languages/elixir/reference/implementing-a-concept-exercise.md

```diff
@@ -1,5 +1,124 @@
-# How to implement an Elixir concept exercise
+# How to implement an Elixir Concept Exercise
 
-TODO: describe how to implement a concept exercise for the Elixir track. For inspiration, check out the [C# version of this file][csharp-implementing].
+This document describes how to implement a Concept Exercise for the Elixir track. As this document is generic, the following placeholders are used:
 
-[csharp-implementing]: ../../csharp/reference/implementing-a-concept-exercise.md
+- `$slug`: the name of the exercise in snake_case (e.g. `anonymous-methods`).
+- `$concepts`: the Concepts the exercise is about (e.g. `loops`),
+- `$concept-1`: a single Concept slug,
+- `$prerequisite-n`: a single Concept slug,
+- `$uuid`: a _new_ v4 UUID (random! and unique to all other concept and practice exercises)
+- `$first-and-last-name`: your first and last name (e.g. `Tim Austin`)
+- `$git-email`: the email address you use for git (e.g. `tim@neenjaw.com`)
+
+Before implementing the exercise, please make sure you have a good understanding of what the exercise should be teaching (and what not). This information can be found in the exercise's GitHub issue. Having done this, please read the [Elixir Concept exercises introduction][concept-exercises]. If you have come up with something completely new, create a new issue _first_ so we can discuss the Concept Exercise.
+
+To implement a Concept Exercise, the following files must be created:
+
+```text
+languages
+└── elixir
+    └── exercises
+        └── concept
+            └── $slug
+                ├── .docs
+                │   ├── instructions.md
+                │   ├── introduction.md
+                │   ├── hints.md
+                │   └── after.md (optional)
+                ├── .meta
+                │   ├── config.json
+                │   └── design.md
+                ├── lib
+                │   ├── example.js
+                │   └── $slug.ex
+                ├── mix.exs
+                ├── mix.lock
+                └── test
+                    ├── $slug_test.exs
+                    └── test_test.exs
```

Comment:
Are there multiple test files?

Reply:
```suggestion
                    └── test_helper.exs
```

Reply:
I made a spelling error, the test helper is generated by the mix project generation.